types/datum.go

```diff
@@ -122,6 +122,11 @@ func (d *Datum) IsNull() bool {
 	return d.k == KindNull
 }
 
+// Size gets the length of byte of the datum.
+func (d *Datum) Size() int {
+	return len(d.b)
```

Comment:
We need to switch on `d.k` to determine the size of the datum.

Reply:
Or if we only care about varlen size, then we don't need this method, use `len(d.GetBytes())` will do.